src/core/common/instance.cpp

```diff
@@ -345,6 +345,7 @@ void Instance::Finalize(void)
     IgnoreError(otIp6SetEnabled(this, false));
     IgnoreError(otLinkSetEnabled(this, false));
 
+    Get<KeyManager>().DeleteSecurityKeys();
```

Comment:
I think this call shouldn't remove persistent keys, right? Maybe make `DeleteSecurityKeys` only remove volatile keys and call `Crypto::Storage::DeletePersistentKeys` in the factory reset / erase methods only?

Reply:
I wondered the same, but looking at the documentation of `otInstanceFinalize`:

Reply:
Ok, maybe @abtink will help with the interpretation, but my understanding was always that `otInstanceFinalize` is used to temporarily release runtime resources taken by OT but settings and persistent data should stay as they are.

Reply:
If that is the case, easily fixable :)

Reply:
No, those were all my comments. PR looks good to me beside that :)

Reply:
Thank you for the review, I will push the documentation updates soon.

Reply:
> Ok, maybe @abtink will help with the interpretation, but my understanding was always that otInstanceFinalize is used to temporarily release runtime resources taken by OT but settings and persistent data should stay as they are.

Yes, That's correct.  `otInstanceFinalize()` should not delete any saved info in non-volatile.  It just stops Thread opration and frees the current OT instance.

- We have `otInstanceErasePersistentInfo()` to erase non-voaltile settings.
- And `otInstanceFactoryReset()` which should do the same and then reset (reset platform and re-init OT instance).

Reply:
Thanks for the explanation @abtink. I have updated the implementation accordingly. Please have a look.
@Damian-Nordic updated the documentation.